Automattic/sensei
# Save the quiz pagination form state between pages

Fixes #4392, #4516

### Changes proposed in this Pull Request

* Replace all quiz pagination links with buttons in order to submit the form and save the state on each page change.
* When completing the form, use the saved answers.

### Testing instructions

* Enable the quiz pagination feature:
```php
add_filter( 'sensei_feature_flag_quiz_pagination', '__return_true' );
```
* Make a quiz with multiple questions.
* Enable the pagination by updating the quiz pagination settings (see #4387).
* Enable the quiz reset setting.
* Enroll in the course.
* Open the quiz in the frontend.
* Answer the first question and go to the next page.
* Go back to the previous page and make sure your answer is saved.
* Make sure the `Save` and `Reset` buttons work as expected.
* Answer some of the questions (not all) and make sure you can complete the quiz.
* Make sure you can see the `Reset` button after completing the quiz.
* Make sure all the answered questions are stored for grading.
* Enable the `Random Question Order` setting, re-take the quiz and make sure everything works.

### Deprecated Code

* `the_question_hidden_fields`

### Screenshots

The pagination numbers style also has changed a bit:

<img width="658" alt="Screenshot on 2021-12-17 at 23-00-44" src="https://user-images.githubusercontent.com/1612178/146607478-276df721-34f6-446c-ad7b-63b2ad1e7863.png">
